test-suite/shared/src/test/scala/org/scalajs/testsuite/niobuffer/BaseBufferTest.scala

```diff
@@ -33,11 +33,14 @@ abstract class BaseBufferTest {
     assertEquals(0, allocBuffer(0).capacity)
 
     expectThrows(classOf[Exception], allocBuffer(-1))
+    expectThrows(classOf[Throwable], allocBuffer(0, -1, 1))
+    expectThrows(classOf[Throwable], allocBuffer(1, 0, 1))
+    expectThrows(classOf[Throwable], allocBuffer(0, 1, 0))
+    expectThrows(classOf[Throwable], allocBuffer(1, 0, 0))
```

Comment:
I believe that it was failing with `UBE` with the wrapped typed arrays.

Furthermore the best I could do is `RuntimeException` as on the JVM the exception can be either 
`IndexOutOfBoundsException` or `IllegalArgumentException`.